pkg/controller/backingstore/backingstore_controller.go

```diff
@@ -66,3 +68,19 @@ func Add(mgr manager.Manager) error {
 
 	return nil
 }
+
+// ModeChangedPredicate will only allow events that changed Status.Mode.ModeCode.
+// This predicate should be used only for BackingsStore objects!
+type ModeChangedPredicate struct {
```

Comment:
Can you please prefix the name with BackingStore because this contains a direct cast to the backing store object and it is not a general-purpose predicate. 

Also, this predicate is checking the status section, can we add this to the name as well?

I am suggesting: `BackingStoreStatusModeCodeChanged`

Reply:
That's a hell of a name!
 
![long-name](https://user-images.githubusercontent.com/1091422/89831106-f5c91f80-db65-11ea-9361-711884a134f4.jpg)

Reply:
I still think the name should explain that this predicate is diverging from the established path (not reconciling status changes - because status should always be updated from the owning controller)